File: 08-test-calendar/R/plot_intraday.R
#!/usr/bin/env Rscript
# SAMPLE twin of plot_intraday.py. Optional until R exists.
# Reads out/intraday_sample.csv; base graphics only (no tidyverse required).

this_dir <- (function() {
  args <- commandArgs(trailingOnly = FALSE)
  hit <- grep("^--file=", args, value = TRUE)
  if (length(hit) == 1) {
    return(dirname(normalizePath(sub("^--file=", "", hit))))
  }
  getwd()
})()
root <- normalizePath(file.path(this_dir, ".."))
csv_path <- file.path(root, "out", "intraday_sample.csv")
if (!file.exists(csv_path)) {
  stop("Missing ", csv_path, " — run generate_intraday.py first.")
}

dat <- read.csv(csv_path, stringsAsFactors = FALSE)
be <- dat$break_even_mer[1]
png_path <- file.path(root, "out", "SAMPLE_intraday_mer_cut_r.png")

png(png_path, width = 1050, height = 740)
par(mfrow = c(2, 1), mar = c(4, 4, 3, 1))
plot(
  dat$hour, dat$spend_usd, type = "h", lwd = 8, col = "#93c5fd",
  xlab = "", ylab = "USD (SAMPLE)",
  main = "SAMPLE: hourly spend vs cash-proxy — one day (R twin)"
)
lines(dat$hour, dat$cash_proxy_usd, col = "#1d4ed8", lwd = 2)
rect(10.5, par("usr")[3], 13.5, par("usr")[4], col = rgb(254 / 255, 202 / 255, 202 / 255, 0.45), border = NA)
legend("topleft", c("Spend", "Cash-proxy"), col = c("#93c5fd", "#1d4ed8"), lwd = c(8, 2), bty = "n")

plot(
  dat$hour, dat$mer, type = "b", col = "#1d4ed8", lwd = 2,
  xlab = "Hour", ylab = "Cash MER", xaxt = "n",
  main = "SAMPLE: 11:00, MER below break-even for 3 hours → cut."
)
axis(1, at = dat$hour, labels = dat$hour_label, las = 2, cex.axis = 0.7)
abline(h = be, col = "#ea580c", lty = 2, lwd = 2)
rect(10.5, par("usr")[3], 13.5, par("usr")[4], col = rgb(254 / 255, 202 / 255, 202 / 255, 0.45), border = NA)
text(11, min(dat$mer) + 0.15, "11:00, MER below break-even for 3 hours → cut.", col = "#991b1b", adj = 0, cex = 0.85)
dev.off()
message("Wrote ", png_path)

File: 08-test-calendar/plot_intraday.py
#!/usr/bin/env python3
"""SAMPLE plot: 11:00, MER below break-even for 3 hours → cut."""
from __future__ import annotations

import subprocess
import sys
from pathlib import Path

import matplotlib

matplotlib.use("Agg")
import matplotlib.pyplot as plt
import pandas as pd

HERE = Path(__file__).resolve().parent
OUT = HERE / "out"
CSV = OUT / "intraday_sample.csv"
PNG = OUT / "SAMPLE_intraday_mer_cut.png"


def ensure_csv() -> None:
    if not CSV.exists():
        subprocess.check_call([sys.executable, str(HERE / "generate_intraday.py")])


def main() -> int:
    ensure_csv()
    df = pd.read_csv(CSV)
    be = float(df["break_even_mer"].iloc[0])
    hours = df["hour"].to_numpy()
    fig, axes = plt.subplots(2, 1, figsize=(10.5, 7.4), sharex=True, constrained_layout=True)

    ax = axes[0]
    ax.bar(hours, df["spend_usd"], color="#93c5fd", width=0.7, label="Hourly spend")
    ax.plot(hours, df["cash_proxy_usd"], color="#1d4ed8", lw=2.0, marker="o", ms=3.5, label="Hourly cash-proxy")
    ax.axvspan(10.5, 13.5, color="#fecaca", alpha=0.55, label="11:00–13:00 kill window")
    ax.set_ylabel("USD (SAMPLE)")
    ax.set_title("SAMPLE: hourly spend vs cash-proxy — one day")
    ax.legend(loc="upper left", fontsize=8)
    ax.grid(True, axis="y", alpha=0.3)

    ax = axes[1]
    ax.plot(hours, df["mer"], color="#1d4ed8", lw=2.0, marker="o", ms=4, label="Cash MER")
    ax.axhline(be, color="#ea580c", ls="--", lw=1.4, label=f"Break-even MER {be:.2f}")
    ax.axvspan(10.5, 13.5, color="#fecaca", alpha=0.55)
    ax.set_xticks(hours)
    ax.set_xticklabels(df["hour_label"], rotation=45, ha="right")
    ax.set_xlabel("Hour (SAMPLE day)")
    ax.set_ylabel("Cash MER")
    ax.set_title("SAMPLE: 11:00, MER below break-even for 3 hours → cut.")
    ax.legend(loc="upper right", fontsize=8)
    ax.grid(True, axis="y", alpha=0.3)
    ax.annotate(
        "11:00, MER below break-even for 3 hours → cut.",
        xy=(11, df.loc[df["hour"] == 11, "mer"].iloc[0]),
        xytext=(2, 1.15),
        textcoords="data",
        fontsize=9,
        color="#991b1b",
        arrowprops=dict(arrowstyle="->", color="#991b1b"),
    )
    fig.text(
        0.01,
        0.01,
        "SAMPLE. Hours, not weeks. Not a bidding bot. Kill is a same-day cut, not a 4-week geo readout.",
        fontsize=8,
        color="#374151",
    )
    fig.savefig(PNG, dpi=140)
    plt.close(fig)
    print(f"Wrote {PNG}")
    return 0


if __name__ == "__main__":
    sys.exit(main())
